test/decorators.test.ts

```diff
@@ -49,6 +51,30 @@ describe('Request decorators', () => {
     expect(login.headers['set-cookie']).toBeUndefined() // no user added to session
   })
 
+  test('logIn allows logging in an arbitrary user for the duration of the request if session=false', async () => {
+    if (process.env.SESSION_PLUGIN !== '@fastify/session') return
```

Comment:
It's better to skip the test instead

Reply:
unlike tap jest doesn't allow you to skip the test once it has started running 😕 . To get it to skip I might have to add some more conditional logic before the test runs.

Reply:
The community has been [asking ](https://github.com/facebook/jest/issues/8604) for a `skip` feature for a long time but jest team does not think skip should ever be necessary :)

Reply:
You can't skip the test once it has started, but you can skip it before it starts with `test.skip("the test name")`. You could do `const sessionOnlyTest = process.env.SESSION_PLUGIN == "@fastify/session" ? test : test.skip` for a nice reusable primitive.